app/main.py

```diff
@@ -1,2 +1,82 @@
 class Dictionary:
-    pass
+    def __init__(self):
+        self.__size = 0
+        self.hash_table = [None] * 8
+        self.capacity = 8
+        self.limit_to_resize = self.capacity * 2 / 3
+
+    def get_need_index(self, ind, key, hash_key):
+        if self.hash_table[ind] is None:
+            return ind
+        if self.hash_table[ind][0] == key:
+            return ind
+
+        while True:
+            if not self.hash_table[ind]:
+                return ind
+            if self.hash_table[ind][0] == key and \
+                    self.hash_table[ind][1] == hash_key:
+                return ind
+            ind = (ind + 1) % self.capacity
+
+    def resize(self):
+        if len(self) >= self.limit_to_resize:
+            cash_hash_table = self.hash_table.copy()
+            self.hash_table = [None] * len(cash_hash_table) * 2
+            self.capacity = len(self.hash_table)
+            self.limit_to_resize = self.capacity * 2 / 3
+            for item in cash_hash_table:
+                if item is not None:
+                    (key, hash_key, value) = item
+                    ind = hash_key % self.capacity
+                    if self.hash_table[ind] is None:
+                        self.hash_table[ind] = (key, hash_key, value)
+                        continue
+                    new_index = self.get_need_index(ind, key, hash_key)
+                    self.hash_table[new_index] = (key, hash_key, value)
+
+    def __setitem__(self, key, value):
+        hash_key = hash(key)
+        try:
+            ind = hash_key % self.capacity
+        except TypeError:
+            return
+        new_index = self.get_need_index(ind, key, hash_key)
+        self.hash_table[new_index] = (key, hash_key, value)
+        self.__size = len([item for item in self.hash_table
+                           if item is not None])
```

Comment:
represent it as counter, rise it by 1 when adding element and reduce by 1 when deleting